backend/tests/test_main.py

```diff
@@ -1,18 +1,16 @@
 import pytest
 from fastapi.testclient import TestClient
 
-from src.main import app, get_db
-from tests.conftest import get_test_db
-
-# Override the get_db dependency with get_test_db
-app.dependency_overrides[get_db] = get_test_db
+from src.crud import create_listing, create_user
+from src.main import app
+from src.schemas import ListingCreate, UserCreate
 
 client = TestClient(app)
 
 API_URL = "http://localhost:8000"
```

Comment:
We should use the client instead of some random endpoint. This is likely just calling your local endpoint.